File: src/codes/Users_Manual/controller_2.py
~from pyautd3 import AUTD3, Controller, Device
~from pyautd3.gain import Focus, FocusOption, Null
~from pyautd3.link.nop import Nop
~autd = Controller.open([AUTD3()], Nop())
~x = 0.0
~y = 0.0
~z = 0.0
def key_map(dev: Device) -> str | None:
    if dev.idx == 0:
        return "null"
    if dev.idx == 1:
        return "focus"
    return None


autd.group_send(
    key_map=key_map,
    data_map={"null": Null(), "focus": Focus(pos=[x, y, z], option=FocusOption())},
)

File: src/codes/Users_Manual/controller_2.cs
~using AUTD3Sharp;
~using AUTD3Sharp.Link;
~using AUTD3Sharp.Gain;
~using AUTD3Sharp.Modulation;
~using AUTD3Sharp.Utils;
~using var autd = Controller.Open([new AUTD3()], new Nop());
~var x = 0.0f;
~var y = 0.0f;
~var z = 0.0f;
autd.GroupSend(dev =>
    {
        return dev.Idx() switch
        {
            0 => "null",
            1 => "focus",
            _ => null
        };
    },
    new GroupDictionary {
        { "null", new Null() },
        { "focus", new Focus(pos: new Point3(x, y, z), option: new FocusOption()) }
    }
);